core-web/libs/portlets/dot-ema/src/lib/feature/dot-ema.component.ts

```diff
@@ -160,6 +149,48 @@ export class DotEmaComponent implements OnInit, OnDestroy {
         });
     }
 
+    /**
+     * Handle the persona selection
+     *
+     * @param {DotPersona} persona
+     * @memberof DotEmaComponent
+     */
+    onPersonaSelected(persona: DotPersona & { pageID: string }) {
+        if (persona.identifier === DEFAULT_PERSONA.identifier || persona.personalized) {
+            this.router.navigate([], {
+                queryParams: {
+                    'com.dotmarketing.persona.id': persona.identifier
+                },
+                queryParamsHandling: 'merge'
+            });
+        } else {
+            this.confirmationService.confirm({
+                header: this.dotMessageService.get('editpage.personalization.confirm.header'),
+                message: this.dotMessageService.get(
+                    'editpage.personalization.confirm.message',
+                    persona.name
+                ),
+                acceptLabel: this.dotMessageService.get('dot.common.dialog.accept'),
+                rejectLabel: this.dotMessageService.get('dot.common.dialog.reject'),
+                accept: () => {
+                    this.personalizeService
+                        .personalized(persona.pageID, persona.keyTag)
+                        .subscribe(); // This does a take 1 under the hood
+
+                    this.router.navigate([], {
+                        queryParams: {
+                            'com.dotmarketing.persona.id': persona.identifier
+                        },
+                        queryParamsHandling: 'merge'
+                    });
```

Comment:
Race condition here... I think we need to redirect after personalization was done. No?

Reply:
Yes! You are right